### ISSUE TITLE
feat(population): persistency metrics — wait time, occurrence probability, cumulative count (V.C)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#112]
## Problem / Request
Tier V.C (`06c_persistency.md`) defines the operational forecasting outputs — expected wait time to the next event `E[Δt | t]`, occurrence probability in a horizon `P(N(t, t+h) ≥ 1)`, expected cumulative count, and their posterior-aware versions — from a fitted intensity `λ(t)`. The intensity functions exist (once the primitives land) but the metrics module (`population.persistency`) and the posterior wrappers are ☐. The integral `∫λ` is closed-form for constant / exponential-decay intensities and needs quadrature otherwise; the design asks for it wrapped once and reused.

## User Story
> As an operator, I want `wait_time(posterior, t_now)` and `occurrence_probability(posterior, t_now, horizon)` with credible intervals, so I can decide whether to task a follow-up observation.

## Proposed API
```python
# population/persistency.py
def expected_wait_time(intensity: str, t: float, **theta) -> jax.Array: ...          # ∫₀^∞ exp(−Λ(t, t+s)) ds
def occurrence_probability(intensity: str, t: float, horizon: float, **theta) -> jax.Array: ...   # 1 − exp(−Λ(t, t+h))
def expected_count(intensity: str, t: float, horizon: float, **theta) -> jax.Array: ...
def posterior_metric(metric: Callable, posterior: PoissonRatePosterior | InhomogeneousIntensityPosterior | TMTPPPosterior, *args, quantiles=(0.05, 0.5, 0.95)) -> dict: ...
```

## Implementation Steps
- [ ] Add `population/persistency.py`; use `intensity.compensator` for `Λ`, with `jax.scipy.integrate.trapezoid` on a fixed log-spaced grid for the wait-time integral
- [ ] `posterior_metric` vmaps a metric over posterior samples and returns quantiles (design: "API returns posterior samples by default")
- [ ] Docstring note on Hawkes: metrics are conditional on history; v1 supports Poisson-family intensities, Hawkes opt-in when it lands

## Definition of Done
- [ ] Homogeneous limit: all metrics match the closed-form exponential formulas to machine precision
- [ ] MC self-consistency: thinning-sampled events from a known `λ(t)` reproduce the empirical wait time within MC error
- [ ] Posterior coverage: 95 % CI on `E[Δt]` contains the truth ~95 % of the time across replicates
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] `tests/population/test_persistency.py::test_homogeneous_closed_form`
- [ ] `tests/population/test_persistency.py::test_mc_self_consistency` (`slow`)
- [ ] `tests/population/test_persistency.py::test_diurnal_sanity`

## Relationships
- Parent (theme epic): #75
- Blocked by: methane_pod primitives in tree (intensity registry / compensator)
